corehq/apps/app_manager/app_translations/generators.py

```diff
@@ -0,0 +1,174 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+import polib
+import datetime
+import tempfile
+
+from django.conf import settings
+from memoized import memoized
+
+from collections import namedtuple, OrderedDict
+from corehq.apps.app_manager.app_translations.const import MODULES_AND_FORMS_SHEET_NAME
+
+Translation = namedtuple('Translation', 'key translation occurrences')
+Unique_ID = namedtuple('UniqueID', 'type id')
+
+
+class POFileGenerator:
+    def __init__(self, domain, app_id, version, key_lang, source_lang, lang_prefix):
+        self.domain = domain
+        self.app_id = app_id
+        self.key_lang = key_lang
+        self.source_lang = source_lang
+        self.lang_prefix = lang_prefix
+        self.translations = OrderedDict()
+        self.version = version
+        self.headers = dict()  # headers for each sheet name
+        self.generated_files = list()  # list of tuples (filename, filepath)
+        self.sheet_name_to_module_or_form_type_and_id = dict()
```

Comment:
I do find all this business around sheets, rows, and column indices pretty confusing.  It's a little hard to follow, particularly since the application and transifex don't have anything to do with spreadsheets - that's just how we happen to have easy access to it, since that already exists.

You can punt on this if you don't think it's worth the effort, but you could modify `expected_bulk_app_sheet_rows` to return a custom data model that's easier to work with.

Reply:
Agree.. but going to skip this for now and try to sneak this in during improvements later.